x-pack/plugins/apm/server/lib/helpers/create_es_client/create_apm_event_client/index.ts

```diff
@@ -79,10 +80,17 @@ export function createApmEventClient({
   return {
     async search<TParams extends APMEventESSearchRequest>(
       params: TParams,
-      { includeLegacyData = false } = {}
+      optionsOrOperationName:
+        | string
+        | { includeLegacyData?: boolean; operationName: string }
```

Comment:
I think this is getting a little weird. I'm not sure it was a good idea in the first place to have a secondary options argument. Why not have everything in `params`? We already have custom properties inside the `apm` like `apm.events`.

Reply:
```json
const params = {
  apm: {
   operationName: 'xyz',
   events: [...],
   includeLegacyData: false
  }
};

apmEventClient.search(params);
```

Reply:
Another options is to keep `operationName` as a separate arg but merge `options` and `params`:

const params = {
  apm: {
   events: [ProcessorEvent.metric],
   includeLegacyData: true
  }
};

// having operationName first seems more logical
apmEventClient.search('xyz', params);
```

Reply:
Agreed - I'll go for your second suggestion as the first one makes it a little harder to share search requests. Once we've removed projections we might want to revisit that decision, but right now it makes it a little harder to have `operationName` in `params.apm` as well.